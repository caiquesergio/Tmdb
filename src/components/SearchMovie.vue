<script setup lang="ts">
import { ref } from 'vue';
import router from '../router';

const searchMovie = ref("");

function searchMoviesInput(name: string) {
  router.push({ name: 'search-movie', params: { name: name } });
}

</script>

<template>
  <div class="flex">
    <input type="text" name="movie" id="movie" v-model="searchMovie"
      class="rounded-full px-3 py-2 bg-white border shadow-sm border-slate-300 w-full sm:text-sm focus:ring-1 text-black"
      placeholder="Buscar por um filme">
    <button @click="searchMoviesInput(searchMovie)"
      class="-ml-8 w-24 rounded-full bg-gradient-to-r from-green-400 to-blue-500 border shadow-sm text-white font-semibold">Buscar</button>
  </div>
</template>
