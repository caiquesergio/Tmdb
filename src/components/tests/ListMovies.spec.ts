import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import SearchMovie from "../SearchMovie.vue";

describe("ListMovies", () => {

    const wrapper = mount(SearchMovie, {});

    it('snapshot UI testing', () => {
        expect(wrapper.text()).toMatchSnapshot()
    })
});