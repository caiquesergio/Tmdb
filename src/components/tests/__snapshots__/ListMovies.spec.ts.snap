// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ListMovies > snapshot UI testing 1`] = `"Buscar"`;
